packages/prebuild-config/src/plugins/unversioned/expo-splash-screen/withIosSplashInfoPlist.ts

```diff
@@ -33,7 +33,9 @@ export function setSplashInfoPlist(
   debug(`isDarkModeEnabled: `, isDarkModeEnabled);
 
   if (isDarkModeEnabled) {
-    const existing = IOSConfig.UserInterfaceStyle.getUserInterfaceStyle(config);
+    // IOSConfig.UserInterfaceStyle.getUserInterfaceStyle(config);
+    // Determine if the user manually defined the userInterfaceStyle incorrectly
+    const existing = config.ios?.userInterfaceStyle ?? config.userInterfaceStyle;
```

Comment:
i think it's preferable to keep `IOSConfig.UserInterfaceStyle.getUserInterfaceStyle` unless we change the default value to `automatic`

Reply:
no because this will always warn the user that they misconfigured their project, we should determine the user' intent (attempting to setup dark mode) and ensure dark mode works, unless specified otherwise manually.